lib/services/authority.go

```diff
@@ -3,14 +3,40 @@ package services
 import (
 	"encoding/json"
 	"fmt"
+	"time"
 
+	"github.com/gravitational/teleport"
 	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/utils"
 
 	"github.com/gravitational/trace"
 	"golang.org/x/crypto/ssh"
 )
 
+// CertParams defines all parameters needed to generate a host certificate.
+type CertParams struct {
+	PrivateCASigningKey []byte
+	PublicHostKey       []byte
+	HostID              string
+	NodeName            string
+	ClusterName         string
+	Roles               teleport.Roles
+	TTL                 time.Duration
+}
+
+func (c CertParams) Check() error {
```

Comment:
I'm not making changes to `CertParams`? If it's a convention to use a pointer receiver for all `Check` methods I can change it.